go-controller/pkg/node/gateway_init.go

```diff
@@ -189,7 +190,32 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 	if needIPv4NextHop || needIPv6NextHop || gatewayIntf == "" {
 		defaultGatewayIntf, defaultGatewayNextHops, err := getDefaultGatewayInterfaceDetails(gatewayIntf, config.IPv4Mode, config.IPv6Mode)
 		if err != nil {
-			return nil, "", err
+			if !(errors.As(err, new(*GatewayInterfaceMismatchError)) && config.Gateway.Mode == config.GatewayModeLocal && config.Gateway.AllowNoUplink) {
```

Comment:
`errors.As` checks whether the error has a specific type, unlike the `Is()`, which examines if it is a particular error value. In our case, the returned GatewayInterfaceMismatchError objects are variable. So it would be better to check the error type, instead of the value.

This article discussed this topic in detail. https://www.digitalocean.com/community/tutorials/how-to-add-extra-information-to-errors-in-go

Reply:
In that case I think implementing an `Is` method for the error is the cleanest option.

I have the feeling that `As` is used to match the error to something that might not be an error type. 

But I might be misunderstanding this. So I will leave it to you.